pageserver/src/tenant.rs

```diff
@@ -2012,10 +2245,13 @@ impl Tenant {
     /// Branch an existing timeline
     async fn branch_timeline(
         &self,
-        src: TimelineId,
+        src_timeline: &Arc<Timeline>,
```

Comment:
This change of the argument from `TimelineId` to `Arc<Timeline>` could probably be done as a separate PR. It' s a tiny change, but it makes sense on its own. I'll extract that.

Reply:
FTR, that is https://github.com/neondatabase/neon/pull/3318